src/api/search_transactions.rs

```diff
@@ -80,12 +88,31 @@ impl MinaMesh {
     Ok(user_commands)
   }
 
-  #[allow(dead_code)]
-  async fn fetch_internal_commands(
+  pub async fn fetch_internal_commands(
     &self,
-    _req: &SearchTransactionsRequest,
-  ) -> Result<Vec<BlockTransaction>, MinaMeshError> {
-    unimplemented!()
+    req: &SearchTransactionsRequest,
+    offset: i64,
+    limit: i64,
+  ) -> Result<Vec<InternalCommand>, MinaMeshError> {
+    let query_params = SearchTransactionsQueryParams::try_from(req.clone())?;
+
+    let internal_commands = sqlx::query_file_as!(
+      InternalCommand,
+      "sql/indexer_internal_commands.sql",
+      query_params.max_block,
+      query_params.transaction_hash,
+      query_params.account_identifier,
+      query_params.token_id,
+      query_params.status as Option<TransactionStatus>,
+      query_params.success_status as Option<TransactionStatus>,
```

Comment:
Why are these `as`es necessary?

Reply:
I'm not sure, but compiler complains without them:
```
unsupported type transaction_status for param #5
```